examples/plugin.cpp

```diff
@@ -0,0 +1,185 @@
+		imu_cam_buffer = NULL;
+    
+    // TODO: read Kimera flag file path from runner and find a better way of passing it to gflag
+
+    kimera_pipeline.registerBackendOutputCallback(
+      std::bind(
+        &kimera_vio::pose_callback,
+        this,
+        std::placeholders::_1
+      )
+    );
+
+		_m_pose->put(
+			new pose_type{
+				.sensor_time = std::chrono::time_point<std::chrono::system_clock>{},
+				.position = Eigen::Vector3f{0, 0, 0},
+				.orientation = Eigen::Quaternionf{1, 0, 0, 0}
+			}
+		);
+
+#ifdef CV_HAS_METRICS
+		cv::metrics::setAccount(new std::string{"-1"});
+#endif
+
+	}
+
+
+	virtual void start() override {
+		plugin::start();
+		sb->schedule<imu_cam_type>(id, "imu_cam", [&](const imu_cam_type *datum) {
+			this->feed_imu_cam(datum);
+		});
+	}
+
+
+	std::size_t iteration_no = 0;
+	void feed_imu_cam(const imu_cam_type *datum) {
+		// Ensures that slam doesnt start before valid IMU readings come in
+		if (datum == NULL) {
+			assert(previous_timestamp == 0);
+			return;
+		}
+
+		// This ensures that every data point is coming in chronological order If youre failing this assert, 
+		// make sure that your data folder matches the name in offline_imu_cam/plugin.cc
+		assert(datum->dataset_time > previous_timestamp);
+		previous_timestamp = datum->dataset_time;
+
+		imu_cam_buffer = datum;
+
+		VIO::Vector6 imu_raw_vals;
+		imu_raw_vals << datum->linear_a.cast<double>(), datum->angular_v.cast<double>();
+
+		// Feed the IMU measurement. There should always be IMU data in each call to feed_imu_cam
+		assert((datum->img0.has_value() && datum->img1.has_value()) || (!datum->img0.has_value() && !datum->img1.has_value()));
+		kimera_pipeline.fillSingleImuQueue(VIO::ImuMeasurement(datum->dataset_time, imu_raw_vals));
+
+		// If there is not cam data this func call, break early
+		if (!datum->img0.has_value() && !datum->img1.has_value()) {
+			return;
+		}
+
+#ifdef CV_HAS_METRICS
+		cv::metrics::setAccount(new std::string{std::to_string(itew_pose_blkf_rot
+		}
+#else
+#warning "No OpenCV metrics available. Please recompile OpenCV from git clone --branch 3.4.6-instrumented https://github.com/ILLIXR/opencv/. (see install_deps.sh)"
+#endif
+
+		cv::Mat img0{*imu_cam_buffer->img0.value()};
+		cv::Mat img1{*imu_cam_buffer->img1.value()};
+		cv::cvtColor(img0, img0, cv::COLOR_BGR2GRAY);
+		cv::cvtColor(img1, img1, cv::COLOR_BGR2GRAY);
+		// VIOParams
+		VIO::CameraParams left_cam_info = kimera_pipeline_params.camera_params_.at(0);
+		VIO::CameraParams right_cam_info = kimera_pipeline_params.camera_params_.at(1);
+		kimera_pipeline.fillLeftFrameQueue(VIO::make_unique<VIO::Frame>(kimera_current_frame_id,
+																	datum->dataset_time,
+																	left_cam_info, img0));
+		kimera_pipeline.fillRightFrameQueue(VIO::make_unique<VIO::Frame>(kimera_current_frame_id,
+																	datum->dataset_time,
+																	right_cam_info, img1));
+
+		kimera_pipeline.spin();
+    std::cout << "SPIN FULL\n";
```

Comment:
Guard with debug. You would have to modify the makefile to set/unset the debug flag.